lib/index.js

```diff
@@ -15,329 +15,303 @@ const internals = {};
 
 exports.genMigrationFile = (options, cb) => {
 
-    Joi.assert(options, Schema.options, 'Bad options passed to schwifty-migration.');
+    const validOptions = Joi.validate(options, Schema.options);
+    if (validOptions.error) {
+        return cb(new Error(`Bad options passed to schwifty-migration: ${validOptions.error.message}`));
+    }
 
     const userOptions = Hoek.shallow(options);
 
     if (userOptions.models.length === 0) {
-        return cb(null, '');
+        return cb(null, 'No models passed');
     }
 
-    const { models, migrationsDir, knex } = userOptions;
-
-    const migrationGroup = { models, migrationsDir, knex };
-
-    // Grabbed from Schwifty's migrator.js
-    // Absolutize and normalize
-
-    migrationGroup.migrationsDir = Path.isAbsolute(migrationGroup.migrationsDir) ?
-        Path.normalize(migrationGroup.migrationsDir) :
-        Path.join(process.cwd(), migrationGroup.migrationsDir);
-
-    // Grab join table info
-
-    let joinTableCollector = [];
+    // Ping the knex instance to make sure we're connected to the db
+    userOptions.knex.queryBuilder().select(userOptions.knex.raw('1'))
+        .then(() => {
 
-    Items.serial(migrationGroup.models, (model, next) => {
+            const { models, migrationsDir, knex } = userOptions;
+            const migrationGroup = { models, migrationsDir, knex };
 
-        internals.getJoinTableInfo(model, knex, (_, info) => {
+            // Grabbed from Schwifty's migrator.js
+            // Absolutize and normalize
 
-            if (info) {
-                joinTableCollector = joinTableCollector.concat(info);
-            }
-            next();
-        });
-    },
-    (_) => {
+            migrationGroup.migrationsDir = Path.isAbsolute(migrationGroup.migrationsDir) ?
+                Path.normalize(migrationGroup.migrationsDir) :
+                Path.join(process.cwd(), migrationGroup.migrationsDir);
 
-        const joinTables = joinTableCollector.reduce((collector, joinTable) => {
+            // Grab join table info
 
-            // Dedupe the joinTables -- multiple models can reference the same joinTable,
-            // so we might have more than one entry for the same join table
-            // across different models.
+            let joinTableCollector = [];
 
-            const { tableName, joiSchema } = joinTable;
+            Items.serial(migrationGroup.models, (model, next) => {
 
-            if (!collector[tableName]) {
+                internals.getJoinTableInfo(model, knex, (ignoreError, info) => {
```

Comment:
I feel pretty sure we shouldn't be ignoring this error—it reaches out to the database, right?

Reply:
Yes there are a few things that reach out to the db that I'm ignoring errors from: Main reason I wanted to move this forward and don't really know how to test those errors

Reply:
You can move forward with missing test coverage, and we can work on how to best test that.  The danger of ignoring errors just for test coverage isn't worth it!

Reply:
Alright 👍

Reply:
I did some pretty wacky stuff in Tandy in order to trigger errors - you might find some inspiration there.